synapse/lib/structlog.py

```diff
@@ -0,0 +1,37 @@
+import json
+
+import logging
+
+class JsonFormatter(logging.Formatter):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def format(self, record: logging.LogRecord):
+
+        record.message = record.getMessage()
+        mesg = self.formatMessage(record)
+        ret = {
+            'message': mesg,
+            'logger': record.name,
+            'thread': record.threadName,
+            'process': record.processName,
+            'filename': record.filename,
+            'level': record.levelname,
+            'func': record.funcName,
+            'time': self.formatTime(record, self.datefmt),
+        }
+        if record.exc_info:
+            ret['exc_info'] = self.formatException(record.exc_info)
```

Comment:
Feels like this might be a use case for the unified exc info serialization format from common